praw/models/reddit/emoji.py

```diff
@@ -162,8 +186,57 @@ def add(self, name, image_path):
             )
         response.raise_for_status()
 
+        data = {
+            "name": name,
+            "s3_key": upload_data["key"],
+            "post_flair_allowed": bool(post_flair),
+            "user_flair_allowed": bool(user_flair),
+            "mod_flair_only": bool(mod_only),
+        }
+
         url = API_PATH["emoji_upload"].format(subreddit=self.subreddit)
-        self._reddit.post(
-            url, data={"name": name, "s3_key": upload_data["key"]}
-        )
+        self._reddit.post(url, data=data)
         return Emoji(self._reddit, self.subreddit, name)
+
+    def update(
+        self,
+        name,
+        post_flair=True,
+        user_flair=True,
+        mod_only=False,
+    ):
+        """Update the permissions of an emoji in this subreddit.
+
+        :param name: The name of the emoji
+        :param post_flair: A ``bool`` value. If ``False``, the emoji will not
+            be available in the emoji selection for post flair.
+            (default: ``True``).
+        :param user_flair: A ``bool`` value. If ``False``, the emoji will not
+            be available in the emoji selection for user flair.
+            (default: ``True``).
+        :param mod_only:  A ``bool`` value. If ``True``, the emoji will only
+            be available for use.
+            (default: ``False``).
+
+        When updating an emoji's permissions, if you don't explicitly pass the
+        desired ``bool`` values for a parameter, it will submit as the default
+        ``bool`` value stated above.
```

Comment:
I suspect this isn't the best design choice. If I call `update` on an emoji and pass only the `name` parameter, the method should respect my existing defaults. Also, the `name` parameter should be optional.

To accomplish this, I suggest changing the method signature to
```python
def update(
    self,
    name=None,
    post_flair=None,
    user_flair=None,
    mod_only=None,
):
```

and then changing the definition of `data` to be 
```python
data = {
    "name": name if name is not None else self.name,
    "post_flair_allowed": bool(post_flair) if post_flair is not None else self.post_flair,
    "user_flair_allowed": bool(user_flair) if user_flair is not None else self.user_flair,
    "mod_flair_only": bool(mod_only) if mod_only is not None else self.mod_only,
}
```

Alternatively, if the `emoji_permissions` endpoint doesn't require all 4 parameters, then you could do

```python
data = dict()
if name is not None:
    data['name'] = name
if post_flair is not None:
    data['post_flair_allowed'] = post_flair
```
et cetera to build up a dictionary that contains only the parameters the user passed.

Reply:
I had the same thought with setting the default to None, but no matter what I did, it still overrode them with the defaults. I couldn't get around it, so I figured it would be simpler to do it this way.

I definitely understand where you're coming from, though. But it just kept resetting them. As far as I can tell, there doesn't seem to be a way to avoid this. I'll double check it to be sure, but I'm pretty sure this is the case. I'm open to being proved wrong, though.

Reply:
I double checked. I was mistaken. I apologize. You're absolutely right. It should definitely be changed.